### PR TITLE
Cleaning unnecessary synchronization cookies

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/client/sandbox/cookie/index.js [hunk 1]
```diff
@@ -172,16 +172,22 @@ export default class CookieSandbox extends SandboxBase {
...
+        const cookies        = nativeMethods.documentCookieGetter.call(this.document);
+        const parsedCookies  = parseClientSyncCookieStr(cookies);
+        const sessionId      = settings.get().sessionId;
+        const cookiesForSync = [];
 
-        for (const parsedCookie of parsedCookies) {
+        for (const outdatedCookie of parsedCookies.outdated)
+            nativeMethods.documentCookieSetter.call(this.document, generateDeleteSyncCookieStr(outdatedCookie));
+
+        for (const parsedCookie of parsedCookies.actual) {
```
Comment (miherlosev) [inline src/client/sandbox/cookie/index.js]: I propose rename `parsedCookie.isServerSync` to `parsedCookie.forServerSync`

File: src/request-pipeline/header-transforms.js [hunk 2]
```diff
@@ -56,28 +56,32 @@ function transformCookie (src, ctx) {
...
         });
 
-    const obsoleteSyncCookies = ctx.req.headers.cookie
-        ? parseClientSyncCookieStr(ctx.req.headers.cookie)
-            .filter(clientCookie => {
+    if (ctx.req.headers.cookie) {
+        const parsedClientSyncCookie = parseClientSyncCookieStr(ctx.req.headers.cookie);
+
+        const obsoleteSyncCookies = parsedClientSyncCookie.outdated
+            .concat(parsedClientSyncCookie.actual.filter(clientCookie => {
```
Comment (miherlosev) [inline src/request-pipeline/header-transforms.js]: It's difficult to read. Split to several variables

File: test/client/fixtures/sandbox/cookie-test.js [hunk 3]
```diff
@@ -89,6 +91,73 @@ test('remove real cookie after browser processing', function () {
     strictEqual(nativeMethods.documentCookieGetter.call(document).indexOf(uniqKey), -1);
 });
 
+module('unnecessary synchronization cookies');
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/cookie-test.js]: `module(conditional cookie synchronization)`

File: test/client/fixtures/sandbox/cookie-test.js [hunk 4]
```diff
@@ -89,6 +91,73 @@ test('remove real cookie after browser processing', function () {
...
+
+test('different path', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/cookie-test.js]: I run tests on local machine and it failed. Should we prepare `document.cookie` before each test (clear or something else)?

**Comment by AndreyBelym** [comment]
@testcafe-build-bot retest

**Pull request merged**